src/mo_frontend/typing.ml

```diff
@@ -1509,13 +1509,13 @@ and infer_exp'' env exp : T.typ =
     t
   | SwitchE (exp1, cases) ->
     let t1 = infer_exp_promote env exp1 in
-    let t = infer_cases env t1 T.Non cases in
+    let t = infer_cases env true t1 T.Non cases in
     if not env.pre then
       coverage_cases "switch" env cases t1 exp.at;
     t
   | TryE (exp1, cases) ->
     let t1 = infer_exp env exp1 in
-    let t2 = infer_cases env T.catch T.Non cases in
+    let t2 = infer_cases env false T.catch T.Non cases in
```

Comment:
I don't quite understand why try and switch are handled differently?

Reply:
I believe I can simplify this, see above. The distinction was actually because try-catch block seemed to get inferred multiple times, some with `env.pre`...